argoverse/evaluation/detection_utils.py

```diff
@@ -0,0 +1,174 @@
+from a bird's-eye view. We prefer these metrics instead of IoU due to the
+increased interpretability of the error modes in a set of detections.
+
+"""
+
+from enum import Enum, auto
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from scipy.spatial.transform import Rotation as R
+
+from argoverse.utils.transform import quat_first2last
+
+
+class SimFnType(Enum):
+    CENTER = auto()
+
+
+class DistFnType(Enum):
+    TRANSLATION = auto()
+    SCALE = auto()
+    ORIENTATION = auto()
+
+
+class InterpType(Enum):
+    ALL = auto()
+
+
+class FilterMetric(Enum):
+    EUCLIDEAN = auto()
+
+
+def filter_instances(
+    instances: np.ndarray,
+    target_class: str,
+    filter_metric: FilterMetric = FilterMetric.EUCLIDEAN,
+    max_dist: float = 50.0,
+) -> np.ndarray:
+    """Filter the annotations based on a set of conditions.
+
+    Args:
+        annos: The instances to be filtered.
+        target_class: The name of the class of interest.
+        filter_metric: The range metric used for filtering.
+
+    Returns:
+        The filtered annotations.
+    """
+    instances = np.array([instance for instance in instances if instance.label_class == target_class])
+
+    if filter_metric == FilterMetric.EUCLIDEAN:
+        centers = np.array([dt.translation for dt in instances])
+        filtered_annos = np.array([])
+
+        if centers.shape[0] > 0:
+            dt_dists = np.linalg.norm(centers, axis=1)
+            filtered_annos = instances[dt_dists < max_dist]
+
+        return filtered_annos
+
+
+def get_ranks(dts: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
+    """Get the rankings for the detections.
+
+    Args:
+        dts: Detections.
+
+    Returns:
+        scores: The detection scores.
+        ranks: The ranking for the detections.
+
+    """
+    scores = np.array([dt.score for dt in dts])
+    ranks = scores.argsort()[::-1]
+    return np.expand_dims(scores, 1)[ranks], ranks
+
+
+def interp(prec: np.ndarray, method: InterpType = InterpType.ALL) -> np.ndarray:
+    """Interpolate the precision over all recall levels.
+
+    Args:
+        prec: Precision at all recall levels.
+        method: Accumulation method.
+
+    Returns:
+        Interpolated precision at all recall levels.
+    """
+    if method == InterpType.ALL:
+        prec_interp = np.maximum.accumulate(prec[::-1])[::-1]
+    else:
+        raise NotImplemented("This interpolation method is not implemented!")
+    return prec_interp
+
+
+def compute_match_matrix(dts: np.ndarray, gts: np.ndarray, metric: SimFnType) -> np.ndarray:
+    """Calculate the match matrix between detections and ground truth labels,
+    using a specified similarity function.
+
+    Args:
+        dts: Detections.
+        gts: Ground truth labels.
+        metric: Similarity metric type.
+
+    Returns:
+        Interpolated precision at all recall levels.
```

Comment:
i don't think this docstring is right. Don't we return `sims`? What is the shape of `sims`?